src/components/icon-button/icon-button.test.tsx

```diff
@@ -0,0 +1,153 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+
+import Icon from "../icon";
+import IconButton from ".";
+import { testStyledSystemSpacing } from "../../__spec_helper__/__internal__/test-utils";
+import Logger from "../../__internal__/utils/logger";
+
+jest.mock("../../__internal__/utils/logger");
+
+test("should render with an `Icon` child", () => {
+  render(
+    <IconButton aria-label="icon-button">
+      <Icon type="bin" />
+    </IconButton>
+  );
+
+  const iconButton = screen.getByRole("button", { name: "icon-button" });
+  const icon = screen.getByTestId("icon");
+
+  expect(iconButton).toBeVisible();
+  expect(iconButton).toContainElement(icon);
+  expect(icon).toBeVisible();
+});
+
+test("sets the 'aria-label' attribute correctly when a custom value is passed to the `aria-label` prop", () => {
+  render(
+    <IconButton aria-label="alternative-aria-label">
+      <Icon type="bin" />
+    </IconButton>
+  );
+
+  const iconButton = screen.getByRole("button", {
+    name: "alternative-aria-label",
+  });
+
+  expect(iconButton).toBeVisible();
+  expect(iconButton).toHaveAccessibleName("alternative-aria-label");
```

Comment:
**comment (non-blocking):** I don't think the accessible name assertion is necessary as  `getByRole()` already queries by the accessible name - see [docs](https://testing-library.com/docs/queries/byrole#:~:text=You%20can%20query,up%20to%20you.)